src/Combinatorics/Matroids/matroid_strata_grassmannian.jl

```diff
@@ -0,0 +1,333 @@
+                                xdict::Dict{Vector{Int64}, MPolyElem}) where T <: MPolyElem
+    
+    Id = identity_matrix(R,d)
+    Xpre = make_coordinate_matrix_no_identity(d, n, MC, R, x, xdict)
+    return interlace_columns(Id, Xpre, B, R, x)
+end
+
+
+
+function bases_determinants(d::Int64, n::Int64, Bs::Vector{Vector{Int64}},
+                            MC::Vector{Vector{Int64}},
+                            B::Vector{Int64}, R::MPolyRing, x::Vector{T},
+                            xdict::Dict{Vector{Int64}, MPolyElem}) where T <: MPolyElem
+    
+    X = make_coordinate_matrix(d, n, MC, B, R, x, xdict)
+    return unique!([det(X[:, b]) for b in Bs ])
+end
+
+
+
+function localizing_semigroup(d::Int64, n::Int64, Bs::Vector{Vector{Int64}},
+                              MC::Vector{Vector{Int64}}, B::Vector{Int64},
+                              R::MPolyRing, x::Vector{T},
+                              xdict::Dict{Vector{Int64}, MPolyElem}) where T <: MPolyElem
+    
+    basesX = bases_determinants(d, n, Bs, MC, B, R, x, xdict)
+    
+    sTotal = MPolyPowersOfElement(basesX[1])
+    
+    if length(basesX) == 1
+        return sTotal
+    end
+    
+    for i in 2:length(basesX)
+        if !(basesX[i] in sTotal)
+            sTotal = product(sTotal, MPolyPowersOfElement(basesX[i]))
+        end
+    end
+    
+    return sTotal
+    
+end
+
+
+function matroid_stratum_matrix_coordinates_given_ring(d::Int64, n::Int64,
+                                                       M::Matroid,
+                                                       F::AbstractAlgebra.Ring,
+                                                       B::Vector{Int64},
+                                                       R::MPolyRing,
+                                                       x::Vector{T},
+                                                       xdict::Dict{Vector{Int64}, MPolyElem}) where T <: MPolyElem
+    
+    Bs = bases(M)
+    NBs = nonbases(M)
+    MC = bases_matrix_coordinates(Bs,B)
+
+
+    S = localizing_semigroup(d, n, Bs, MC, B, R, x, xdict)
+    SinvR , iota = Localization(R, S)
+    X = make_coordinate_matrix(d, n, MC, B, R, x, xdict)
+
+    Igens = unique!([det(X[:, nb]) for nb in NBs ])
+
+    if length(Igens) == 0
+        return (X, SinvR, iota(ideal(R, [])))
+    else
+        Ipre = ideal(SinvR, Igens)
+        return (X, SinvR, Ipre)
+    end
+
+end
+
+@doc Markdown.doc"""
+    matroid_stratum_matrix_coordinates(M::Matroid, B::Vector{Int64}, F::AbstractAlgebra.Ring = ZZ)
+
+Return the data of the coordinate ring of the matroid stratum of M in the Grassmannian with respect to matrix coordinates. Here, ```B``` is a basis of ```M``` and the submatrix with columns indexed by ```B''' is the identity. This function returns a pair ```(SinvR, I)``` where the coordinate ring is isomorphic to ```SinvR / I```.
+
+# Examples
+```jldoctest
+julia> M = fano_matroid()
+julia> matroid_stratum_matrix_coordinates(M, [1,2,4], GF(2))
```

Comment:
```suggestion
julia> M, A, B = matroid_stratum_matrix_coordinates(M, [1,2,4], GF(2));
```
and then show `M`, `A`, and `B` one after the other.